holoviews/core/spaces.py

```diff
@@ -1389,6 +1388,51 @@ def _cache(self, key, val):
         self[key] = val
 
 
+    def apply(self, function, streams=[], link_inputs=True, dynamic=None, **kwargs):
+        """Applies a function to all (Nd)Overlay or Element objects.
+
+        Any keyword arguments are passed through to the function. If
+        keyword arguments are instance parameters, or streams are
+        supplied the returned object will dynamically update in
+        response to changes in those objects.
+
+        Args:
+            function: A callable function
+                The function will be passed the return value of the
+                DynamicMap as the first argument and any supplied
+                stream values or keywords as additional keyword
+                arguments.
+            streams (list, optional): A list of Stream objects
+                The Stream objects can dynamically supply values which
+                will be passed to the function as keywords.
+            link_inputs (bool, optional): Whether to link the inputs
+                Determines whether Streams and Links attached to
+                original object will be inherited.
+            dynamic (bool, optional): Whether to make object dynamic
+                By default object is made dynamic if streams are
+                supplied, an instance parameter is supplied as a
+                keyword argument, or the supplied function is a
+                parameterized method.
+            kwargs (dict, optional): Additional keyword arguments
+                Keyword arguments which will be supplied to the
+                function.
+
+        Returns:
+            A new object where the function was applied to all
+            contained (Nd)Overlay or Element objects.
+        """
+        if dynamic == False:
+            samples = tuple(d.values for d in self.kdims)
+            if not all(samples):
+                raise ValueError('Applying a function to a DynamicMap '
+                                 'and setting dynamic=False is only '
+                                 'possible if key dimensions define '
+                                 'a discrete parameter space.')
+            return HoloMap(self[samples]).apply(
+                function, streams, link_inputs, dynamic, **kwargs)
+        return super(DynamicMap, self).apply(function, streams, link_inputs, dynamic, **kwargs)
+
+
     def map(self, map_fn, specs=None, clone=True, link_inputs=True):
```

Comment:
Don't we still prefer `ViewableElement` as the default `spec` for `.map`?

Reply:
I see little reason to use ``.map`` in this scenario but I guess it's still more useful than the default.

Reply:
I meant that was an orthogonal change that did seem better than the old default. Might as well update the default in this PR imho as this is where the idea was first introduced...

Reply:
New suggestion is mentioned in https://github.com/pyviz/holoviews/issues/3544